### PR TITLE
Removing mentions of .done and replacing with .toPromise()

### PR DESCRIPTION
**Pull request by JSoet**
Just updating the readme documentation

### CODE REVIEW

File: docs/api/README.md [hunk 1]
```diff
@@ -1244,7 +1244,7 @@ test('my oddOrEven saga', assert => {
     );
 
     a.equal(
-      data.gen.next().done,
+      data.gen.next().toPromise(),
```
Comment (restrry) [inline docs/api/README.md]: that's iterator's flag showing whether the iterator is finished or not. so no needs for changes

Comment (Andarist) [reply docs/api/README.md]: @JSoet would u like to remove those changes here? The only needed change is [this one](https://github.com/redux-saga/redux-saga/pull/1456/files#diff-0f72f7af9b98c6cb0c52d489836df1e8R965).

**Comment by JSoet** [comment]
Yep, I will but I'm on vacation for the next week and didn't get to this
before leaving, sorry, and I don't think I can make the changes on my
mobile, so if it can wait until I'm back then I'll do it.

**Comment by Andarist** [comment]
No problem, we are not in a rush with this. Enjoy your vacation 👍

**Comment by JSoet** [comment]
K, I changed the other 2 instances back, I assume they were both iterators then. I also noticed a typo in the test output above so changed that while I was at it. Sorry for the delay and the original mistake, and thanks for making a great library :)

**Comment by Andarist** [comment]
Thanks!

**Pull request merged**